src/Onnx/OnnxToFeld.hs

```diff
@@ -202,15 +219,26 @@ mkAccess ti = vname <> " = sel (Proxy @" <> show (tiField ti) <> ") weights ! (Z
   where vname = mangle $ tiName ti
 
 -- | Constuct the C code to read initialized tensors
-mkInitReadFile :: FilePath -> [[TensorInfo]] -> L.ByteString
-mkInitReadFile hf initGroups = L.concat $ start : map initVec initGroups ++ map initTensor (concat initGroups) ++ [end]
+mkInitReadFile :: FilePath -> L.ByteString -> [[TensorInfo]] -> L.ByteString
+mkInitReadFile hf weightRecTC initGroups = L.concat $ start : map initVec initGroups ++ map initTensor (concat initGroups) ++ [end]
   where start = L.unlines
                 [ "#include \"" <> fromString hf <> "\""
                 , ""
                 , "#include <stdlib.h>"
                 , "#include <stdio.h>"
                 , "#include <inttypes.h>"
                 , ""
+                , "static void check(int n, int ref)"
+                , "{"
+                , "  static int items = 0;"
+                , "  if( n != ref ) {"
```

Comment:
`if (n != ref) {` is mostly the style for the runtime system code.

Reply:
All right.